### ISSUE TITLE
ON / OFF status in real time

### ISSUE THREAD

**Issue by SmartM-ui** [fuatakgun/eufy_security#62]
Hi @fuatakgun 
I noticed that if I change the on off status of the cam through the original app, in the home assistant, the status change is displayed with about 10 minutes delay.

To have it updated in real time, the security status must be changed, via the official app, by setting from home to away, or via the alarm activation service in the home assistant.

Conversely, I tried to turn the cam on and off from Home Assistant and the official app immediately shows the status change, closing and reopening the Eufy app.

Would it be possible to create the service to get the on / off status in real time?

Thanks

**Comment by fuatakgun** [comment]
As we have covered in previous issue, fetching on/off status real time is not feasible, given that eufy is not sending push notifications to its clients about state changes. My proposal is to create a service to fetch latest status and you can use this service as you want.

`Conversely, I tried to turn the cam on and off from Home Assistant and the official app immediately shows the status change, closing and reopening the Eufy app.` As you can see here, it actually did not show up immediately, you had to reopen the app, which had triggered to fetch latest status. Please do not mix it.

**Comment by SmartM-ui** [comment]
> As we have covered in previous issue, fetching on/off status real time is not feasible, given that eufy is not sending push notifications to its clients about state changes. My proposal is to create a service to fetch latest status and you can use this service as you want.
> 
> `Conversely, I tried to turn the cam on and off from Home Assistant and the official app immediately shows the status change, closing and reopening the Eufy app.` As you can see here, it actually did not show up immediately, you had to reopen the app, which had triggered to fetch latest status. Please do not mix it.

OK, 
if you can create the service it would be a useful option!
Thanks

**Comment by SmartM-ui** [comment]
> replace `garage_camera` with your own camera name
> 
> ```
> {{ state_attr("camera.garage_camera", "data")["statusled"] }}
> ```

Thanks @fuatakgun for that too.
I noticed some strange behavior!

From the developer tools and in the camera.eufy attributes, the status is changed immediately.

The sensor created, however, is not always updated:

```
sensor:
  - platform: template
    sensors:
      eufy_status_led:
        friendly_name: "Eufy Status Led"
        value_template: "{{ state_attr('camera.xxx', 'data')['statusLed'] }}"
```

Look at the attribute value (False) and the state of the new sensor (True): 

<img width="665" alt="Schermata 2021-10-06 alle 17 40 08" src="https://user-images.githubusercontent.com/63554131/136237812-b8914467-7f87-4cc6-af7b-d131b2c9d699.png">

**Comment by fuatakgun** [comment]
Check this line: https://github.com/fuatakgun/eufy_security/blob/19fe9d924358c2b2f0569085167521f202725afb/custom_components/eufy_security/binary_sensor.py#L24

You see there is a list of binary sensors here. Just add all applicable binary sensors here and they will be generated as a real sensor in your machine. After testing out locally, create a push request so we can merge your change into codebase.

You can go ahead and change custom_components/eufy_security/binary_sensor.py and restart your HA to pick up changes and create respective sensors.

**Comment by SmartM-ui** [comment]
Hi @fuatakgun 
I added more binary_sensors and this is the result:

```
         ("autonightvision", "auto Night vision", "autoNightvision", None, DEVICE_CLASS_MOTION),
```
Binary sensor imported as entity = OK
Status change displayed in real time after starting the service "driver.poll_refresh" = OK

```
         ("microphone", "Microphone", "microphone", None, DEVICE_CLASS_MOTION),
```
Binary sensor imported as entity = OK
Status change displayed in real time after starting the service "driver.poll_refresh" = NO

```
         ("statusled", "Status Led", "statusLed", None, DEVICE_CLASS_MOTION),
```
Binary sensor imported as entity = NO

**Comment by SmartM-ui** [comment]
> Hey, it should be `ledStatus` in our existing version. You can validate the attribute names using state dumper in developer settings page.

Aaargh!
I had written the name of the binary sensor that I had tried to create myself with the template! :-)

OK, he also sees the led. Thanks

I have entered all the binary sensors available, tomorrow I will check who is able to update in real time after running the service. 

<img width="341" alt="Schermata 2021-10-07 alle 01 16 05" src="https://user-images.githubusercontent.com/63554131/136296324-4ca1ba72-17fc-4331-8664-56b8b87c2343.png">

If tomorrow you explain to me how to create the push request, after testing the sensors, I will add the request

Good night

**Comment by SmartM-ui** [comment]
> apparently, this is the bug which was resolved in 0.4.0 version of add-on but we are not able to update into that one yet because of p2p live stream problem. If you do not have a direct dependency on p2p live stream, go ahead and update your add on using following this: #59 and stick with `statusLed`, apparently `ledStatus` is deprecated

Hi @fuatakgun 
I have configured the add-on:
Eufy Security WS Addon
Current version: 0.4.2

The "ledStatus" sensor is never displayed both in the camera.salotto and in the binary_sensor.salotto_enabled

The "statusLed" sensor attribute is displayed only after a change from the official app of the status LED

I tried reporting in the binary_sensor.py
```
("statusled", "Status Led", "statusLed", "mdi: led-on", None),
```
but the sensor is not created, as was the case with the previous addon

In developer templates / tools, when the sensor is displayed as an attribute after an app change, it is reported correctly
```
{{state_attr ('room.salotto', 'data') ['statusLed']}}: True

{{state_attr ('binary_sensor.salotto_enabled', 'statusLed')}}: True
```

Can I perform any other tests?

**Comment by DivanX10** [comment]
Yes, indeed, the status changes faster when they are in the file binary_sensor.py . Then it would be good for the author to make a file switch.py with a standard set of switches, but in order to be able to add your own switches, for example, the author will indicate all possible switches, but comment them out, and we just need to uncomment the necessary switches

**Comment by SmartM-ui** [comment]
@DivanX10 
If you are interested, I had also created switches with the service
https://github.com/fuatakgun/eufy_security/issues/63#issuecomment-934423337

**Comment by fuatakgun** [comment]
service is created to trigger `pull from cloud` and it is named as `force_sync`
you can also configure the interval to automatically fetch the data from cloud.

![image](https://user-images.githubusercontent.com/11085566/136705491-050014a2-b2ac-430f-bfb3-3749c943884a.png)

**Comment by SmartM-ui** [comment]
> service is created to trigger `pull from cloud` and it is named as `force_sync` you can also configure the interval to automatically fetch the data from cloud.
> 
> ![image](https://user-images.githubusercontent.com/11085566/136705491-050014a2-b2ac-430f-bfb3-3749c943884a.png)

Excellent possibility to choose the scan time!

I can't find a way to automate the action on viewing a particular Home Assistant page.
Do you know which command to give to trigger an atomation on the page change?
